shell/edit/kubevirt.io.virtualmachine/VirtualMachineVolume/type/__tests__/HarvesterEditVolume.test.ts

```diff
@@ -0,0 +1,76 @@
+import { mount } from '@vue/test-utils';
+import HarvesterEditVolume from '@shell/edit/kubevirt.io.virtualmachine/VirtualMachineVolume/type/volume.vue';
+import { _EDIT } from '@shell/config/query-params';
+import vSelect, { VueSelectMethods } from 'vue-select';
+
+describe('component: HarvesterEditVolume', () => {
+  it('should display all the inputs', () => {
+    const wrapper = mount(HarvesterEditVolume, {
+      propsData:  { validateRequired: true },
+      mocks:     {
+        $store:    {
+          getters: {
+            'harvester/all': jest.fn(),
+            'i18n/t':                  jest.fn(),
+            'i18n/exists':             jest.fn(),
+          }
+        },
+      }
+    });
+
+    const inputWraps = wrapper.findAll('[data-testid^=input-harvester-volume-]');
+
+    expect(inputWraps).toHaveLength(4);
+  });
+
+  it.each([
+    'name',
+    'size',
+  ])('should emit an update on %p input blur', (field) => {
+    const wrapper = mount(HarvesterEditVolume, {
+      propsData:  { mode: _EDIT, validateRequired: true },
+      mocks:     {
+        $store:    {
+          getters: {
+            'harvester/all': jest.fn(),
+            'i18n/t':                  jest.fn(),
+            'i18n/exists':             jest.fn(),
+          }
+        },
+      }
+    });
+    const input = wrapper.find(`[data-testid="input-harvester-volume-${ field }"]`).find('input');
+    const newValue = 123;
+
+    input.setValue(newValue);
+    input.trigger('blur');
+
+    expect(wrapper.emitted('update')).toHaveLength(1);
+  });
+
+  it.each([
+    'type',
+    'bus'
+  ])('should emit an update on %p selection change', async(field) => {
+    const wrapper = mount(HarvesterEditVolume, {
+      propsData:  { mode: _EDIT, validateRequired: true },
+      mocks:     {
+        $store:    {
+          getters: {
+            'harvester/all': jest.fn(),
+            'i18n/t':                  jest.fn(),
+            'i18n/exists':             jest.fn(),
+          }
+        },
+      }
+    });
+    const select = wrapper
+      .find(`[data-testid="input-harvester-volume-${ field }"]`)
+      .find(vSelect);
+
+    // Component is not in Typescript
+    (select.vm as any as VueSelectMethods).select('whatever');
```

Comment:
I just found out one clashing case with Typescript, which hopefully should not cause so many headaches.
If you want to access to the methods of a Vue component which is not in TS, you will have to override the type in a bad way :grimacing: 
    (select.vm as any as VueSelectMethods).select('whatever');
This is also because for some reasons it's not possible to just set the value for the `v-select` and get an `@input` event 🤔

Reply:
Ok this has been replaced with bare mouse/key events.
- It is not reliable as it allows to set value of disabled input
- It's required to be used `findComponent()` for these cases, but as chain to `.find()` methods it returns error as deprecated in Vue3